lesson25/index.js

```diff
@@ -0,0 +1,136 @@
+
+const openModal = () => {
+    document.getElementById("js-modal-area").classList.add("is-active");
+    bodyElement.classList.add("fixed");
+    document.getElementById("js-modal-contents").scrollTop = 0;
+};
+
+const closeModal = () => {
+    document.getElementById("js-modal-area").classList.remove("is-active");
+    bodyElement.classList.remove("fixed");
+};
+
+checkboxLink.addEventListener("click", openModal);
+checkboxLink.addEventListener("keypress", openModal);
+
+modalCloseButton.addEventListener("click", () => {
+    closeModal();
+    checkFormValidityToEnableSubmitButton();
+});
+
+//モーダル以外の部分を押すとモーダルが閉じる
+document.addEventListener("click", (e) => {
+    if(e.target.classList.contains("js-modal")){
+        closeModal();
+        checkFormValidityToEnableSubmitButton();
+    }
+});
+
+//スクロールが一番下に行ったらチェックボックスをcheckedにする
+const observerTarget = document.getElementById("js-last-sentence");
+const observerOptions = {
+    root: modalInner,
+    threshold: 1.0
+};
+const checkbox = document.getElementById("js-checkbox");
+
+const setCheckedAttributeToCheckbox = ([entry]) => {
+    if (entry.isIntersecting) {
+        checkbox.checked = true;
+        checkbox.disabled = false;
+    }
+};
+
+const observer = new IntersectionObserver(setCheckedAttributeToCheckbox, observerOptions);
+observer.observe(observerTarget);
+
+
+const nameOfInput = document.querySelector(".js-form-name");
+const emailOfInput = document.querySelector(".js-form-email");
+const passwordOfInput = document.querySelector(".js-form-password");
+const formElements = [nameOfInput, emailOfInput, passwordOfInput];
+
+const validationTerms = {
+    name: {
+        upperLimitOfText: 15
+    },
+    password: {
+        pattern: /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z\d]{8,}$/
+    }
+}
+
+const validationOptions = {
+    name: {
+        isValid: () => {
+            return nameOfInput.value.length < validationTerms.name.upperLimitOfText;
+        },
+        errorMessage: "ユーザー名は15文字以内で入力してください",
+    },
+    email: {
+        isValid: () => {
+            return emailOfInput.validity.valid;
+        },
+        errorMessage: "メールアドレスの形式になっていません",
+    },
+    password: {
+        isValid: () => {
+            return validationTerms.password.pattern.test(passwordOfInput.value);
+        },
+        errorMessage: "8文字以上の大小の英数字を交ぜたものにしてください",
+    }
+};
+
+const addInvalidClass = target => target.classList.add("invalid");
+const removeInvalidClass = target => target.classList.remove("invalid");
+const showErrorMessage = target => target.nextElementSibling.textContent = validationOptions[target.id].errorMessage;
+const isNotEmptyOfInput = target => target.value.trim() !== "";
+const isValidFormInput = target => validationOptions[target.id].isValid();
+
+const isValidInputAndCheckbox = () => {
+    const invalidItem = document.getElementsByClassName("invalid");
+    return invalidItem.length === 0 && checkbox.checked;
+};
```

Comment:
nits
I think this function is returns the boolean value of whether the check box is checked, 
so what about the function name as below?

```suggestion
const isCheckedCheckBox = () => {
    const invalidItem = document.getElementsByClassName("invalid");
    return invalidItem.length === 0 && checkbox.checked;
};
```

Reply:
Thanks for the review!
This function contains validation for each input as well as the checkboxes.
Therefore, would you like to keep this name?
If you have any suggestions for improvement, please let me know :)